homeassistant/core.py

```diff
@@ -845,15 +846,21 @@ def async_fire(
         if not listeners:
             return
 
-        for job, event_filter in listeners:
+        for job, event_filter, run_immediately in listeners:
```

Comment:
An alternative approach is to `call_soon` the calling of the listeners, and then use `self.hass.async_run_hass_job`

Reply:
I was going to do that but it would change the order for ones that need to fire from another on firing other ones.  The tests fail when I did